presto-main/src/main/resources/webapp/src/components/LivePlan.jsx

```diff
@@ -125,10 +125,20 @@ class PlanNode extends React.Component<PlanNodeProps, PlanNodeState> {
     }
 
     render() {
+        // get join distribution type by matching details to a regular expression
+        const details = this.props.details;
+        var distribution = "";
+        var distributionRE = /Distribution:\s(REPLICATED|PARTITIONED)/;
```

Comment:
Nice.  Use a more generic regex (something like `[a-zA-Z]+`) to match the distribution type (no need to hard-code the possible values).  And you can inline the regex.

Reply:
```
var distributionPattern = /Distribution:\s+(\w+)/;
```

- var name
- `\w+` -- since we use enum name here, an `_`  or a digit could also appear here
- generally, when you allow _some space_ (`\s`), you either mean "exactly space" (` `), or "some amount of whitespace" (`\s+`)